internal/db/schema/manager.go

```diff
@@ -20,43 +21,54 @@ type driver interface {
 	Lock(context.Context) error
 	Unlock(context.Context) error
 	UnlockShared(context.Context) error
-	// Either starts a transactioon internal to the driver or sets a dirty
-	// bit so if the Run fails the CurrentState reflects it.
+	// StartRun begins a transaction internal to the driver.
 	StartRun(context.Context) error
-	// Either commits the transaction or clears the dirty bit.
-	CommitRun() error
-	// Performs the mutation on the driver.  This should always be
-	// wrapped by StartRun and CommitRun.  The driver must properly
-	// handle the transaction or dirty bit in case of error when
-	// executing Run.
-	Run(context.Context, io.Reader, int) error
+	// CommitRun commits a transaction, if there is an error it should rollback the transaction.
+	CommitRun(context.Context) error
+	// Run will apply a migration. The io.Reader should provide the SQL
+	// statements to execute, and the int is the version for that set of
+	// statements. This should always be wrapped by StartRun and CommitRun.
+	Run(ctx context.Context, migration io.Reader, version int, edition string) error
+	// CurrentState returns the state of the given edition.
+	// ver is the current migration version number as recorded in the database.
 	// A version of -1 indicates no version is set.
-	CurrentState(context.Context) (ver int, everRan bool, dirty bool, err error)
+	// initialized will be true if the schema was previously initialized.
+	CurrentState(ctx context.Context, edition string) (version int, initialized bool, err error)
+	// EnsureVersionTable ensures that the table used to record the schema versions for each edition
+	// exists and is in the correct state.
 	EnsureVersionTable(ctx context.Context) error
+	// EnsureMigrationLogTable ensures that the table used to record migration lgos
+	// exists and is in the correct state.
+	EnsureMigrationLogTable(ctx context.Context) error
+	// GetMigrationLog will retrieve the migration logs from the db for the last
+	// migration.
+	//  The WithDeleteLog option is supported and will remove all log entries,
+	// after reading the entries, when provided.
+	GetMigrationLog(ctx context.Context, opt ...log.Option) ([]*log.Entry, error)
 }
 
 // Manager provides a way to run operations and retrieve information regarding
 // the underlying boundary database schema.
 // Manager is not thread safe.
 type Manager struct {
-	db              *sql.DB
-	driver          driver
-	dialect         string
-	migrationStates map[string]migrationState
+	db       *sql.DB
+	driver   driver
+	dialect  string
+	editions edition.Editions
 }
 
 // NewManager creates a new schema manager. An error is returned
 // if the provided dialect is unrecognized or if the passed in db is unreachable.
 func NewManager(ctx context.Context, dialect string, db *sql.DB, opt ...Option) (*Manager, error) {
```

Comment:
`edition.Dialect` can not be used here since `edition` is an internal package, so it would make this difficult to use since `NewManager` is exported. It could be a `schema.Dialect` though, and probably should be. I can updated that.